x/epochs/abci.go

```diff
@@ -20,39 +20,42 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
 		shouldInitialEpochStart := !epochInfo.EpochCountingStarted && !epochInfo.StartTime.After(ctx.BlockTime())
 
 		epochEndTime := epochInfo.CurrentEpochStartTime.Add(epochInfo.Duration)
-		shouldEpochStart := ctx.BlockTime().After(epochEndTime) && !shouldInitialEpochStart && !epochInfo.StartTime.After(ctx.BlockTime())
-
-		if shouldInitialEpochStart || shouldEpochStart {
-			epochInfo.CurrentEpochStartHeight = ctx.BlockHeight()
-
-			if shouldInitialEpochStart {
-				epochInfo.EpochCountingStarted = true
-				epochInfo.CurrentEpoch = 1
-				epochInfo.CurrentEpochStartTime = epochInfo.StartTime
-				logger.Info(fmt.Sprintf("Starting new epoch with identifier %s epoch number %d", epochInfo.Identifier, epochInfo.CurrentEpoch))
-			} else {
-				ctx.EventManager().EmitEvent(
-					sdk.NewEvent(
-						types.EventTypeEpochEnd,
-						sdk.NewAttribute(types.AttributeEpochNumber, fmt.Sprintf("%d", epochInfo.CurrentEpoch)),
-					),
-				)
-				k.AfterEpochEnd(ctx, epochInfo.Identifier, epochInfo.CurrentEpoch)
-				epochInfo.CurrentEpoch += 1
-				epochInfo.CurrentEpochStartTime = epochInfo.CurrentEpochStartTime.Add(epochInfo.Duration)
-				logger.Info(fmt.Sprintf("Starting epoch with identifier %s epoch number %d", epochInfo.Identifier, epochInfo.CurrentEpoch))
-			}
-			k.SetEpochInfo(ctx, epochInfo)
+		shouldEpochStart := (ctx.BlockTime().After(epochEndTime) && !epochInfo.StartTime.After(ctx.BlockTime())) || shouldInitialEpochStart
+
+		if !shouldEpochStart {
+			return false
+		}
+		epochInfo.CurrentEpochStartHeight = ctx.BlockHeight()
+
+		if shouldInitialEpochStart {
+			epochInfo.EpochCountingStarted = true
+			epochInfo.CurrentEpoch = 1
+			epochInfo.CurrentEpochStartTime = epochInfo.StartTime
+			logger.Info(fmt.Sprintf("Starting new epoch with identifier %s epoch number %d", epochInfo.Identifier, epochInfo.CurrentEpoch))
+		} else {
 			ctx.EventManager().EmitEvent(
 				sdk.NewEvent(
-					types.EventTypeEpochStart,
+					types.EventTypeEpochEnd,
```

Comment:
This is the git diff aiming to make minimal diffs :/

The start event is below. This is the same as the old event.